internal/pkg/cli/init.go

```diff
@@ -187,6 +187,10 @@ func newInitOpts(vars initVars) (*initOpts, error) {
 	}
 	fs := &afero.Afero{Fs: afero.NewOsFs()}
 	cmd := exec.NewCmd()
+	wsAppName := aws.String(tryReadingAppName())
+	if aws.StringValue(wsAppName) == "" {
+		wsAppName = nil
```

Comment:
Hmm I agree that this is the most convenient but it's not clear 🤔 

What if we did `initOpts.initEnvCmd.wsAppName = tryReadingAppName()` in the CLI before calling `Ask` or `Execute`?

Reply:
This probably doesn't work because service init's `Ask` happens before app init's `Execute` 🤔

Reply:
I think I might need to put more comments there specifying `wsAppName` is a pointer string with three valid states (nil, empty and nonempty)

Reply:
Or
```
var wsAppName *string
if ws := tryReadingAppName(); ws != "" {
    wsAppName = aws.String(ws)
} // NOTE: otherwise, pass `nil` to signal that the ws is pending creation.
```

Reply:
Ok i took wx's suggestion and split into two vars.

Reply:
I still think both solutions (a `nil` pointer or a `bool`) is solving the problem in the reverse direction. We're making changes to `job init` and `svc init` so that it works well with `init`, ideally the commands shouldn't have to be adapted for a specific client.

Would the problem go away if `init` had it's own `Validate` and `Ask` methods without depending on any commands? So that we can load up all questions we care about and then create the workload commands objects.